### DISCUSSION TITLE
Python coding : get string between two characters

### DISCUSSION THREAD

**Discussion by FstudentC** [community/community#21422] [Other Feature Feedback, Questions, & Ideas]
<p>I want to get a substring between two characters (/ and ?) from some url I have<br>
For example, I have : <a href="https://partners.doctolib.fr/vaccination-covid-19/rennes/centre-de-vaccination-covid-19-rennes-school-of-business-rennes?pid=practice-205813&amp;enable_cookies_consent=1" rel="noopener nofollow ugc">https://partners.doctolib.fr/vaccination-covid-19/rennes/centre-de-vaccination-covid-19-rennes-school-of-business-rennes?pid=practice-205813&amp;enable_cookies_consent=1</a> and I need to have just “centre-de-vaccination-covid-19-rennes-school-of-business-rennes”</p>
<p>I tried with split(’/, ?’) and split(’/’, ‘?’) but it doesn’t work</p>

**Comment by airtower-luna** [comment, marked as answer]
<p>Take a look at <a href="https://docs.python.org/3/library/re.html">regular expressions (the <code>re</code> module)</a>. They allow you to specify a pattern and match a string or parts of it against that pattern:</p>
<pre><code class="lang-auto">&gt;&gt;&gt; import re
&gt;&gt;&gt; url = 'https://partners.doctolib.fr/vaccination-covid-19/rennes/centre-de-vaccination-covid-19-rennes-school-of-business-rennes?pid=practice-205813&amp;enable_cookies_consent=1'
&gt;&gt;&gt; r = re.compile(r'/([^/\?]+)\?')
&gt;&gt;&gt; m = r.search(url)
&gt;&gt;&gt; m.group(1)
'centre-de-vaccination-covid-19-rennes-school-of-business-rennes'
</code></pre>
<p>The exact pattern depends on what exactly you’re looking for, the one I wrote searches for a <code>/</code>, a group of one or more characters that aren’t <code>/</code> or <code>?</code>, and finally a <code>?</code>. The group is defined by parentheses, and is helpful because you can access matched groups in the Match object (<code>m</code> here).</p>

**Comment by fire-eggs** [comment]
<pre><code class="lang-python">text = "https://partners.doctolib.fr/vaccination-covid-19/rennes/centre-de-vaccination-covid-19-rennes-school-of-business-rennes?pid=practice-205813&amp;enable_cookies_consent=1"

parts1 = text.split('?')
parts2 = parts1[0].split('/')
print(parts2[-1])
</code></pre>
